scripts/EFI_workflow.R

```diff
@@ -0,0 +1,242 @@
+if (args$continue && file.exists(status_file)) {
+  file.remove(status_file)
+}
+
+# Do conversions
+#settings <- PEcAn.workflow::do_conversions(settings)
+
+# Write model specific configs
+if (PEcAn.utils::status.check("CONFIG") == 0) {
+  PEcAn.utils::status.start("CONFIG")
+  settings <-
+    PEcAn.workflow::runModule.run.write.configs(settings)
+  
+  PEcAn.settings::write.settings(settings, outputfile = "pecan.CONFIGS.xml")
+  PEcAn.utils::status.end()
+} else if (file.exists(file.path(settings$outdir, "pecan.CONFIGS.xml"))) {
+  settings <- PEcAn.settings::read.settings(file.path(settings$outdir, "pecan.CONFIGS.xml"))
+}
+
+if ((length(which(commandArgs() == "--advanced")) != 0)
+    && (PEcAn.utils::status.check("ADVANCED") == 0)) {
+  PEcAn.utils::status.start("ADVANCED")
+  q()
+}
+
+# Start ecosystem model runs
+if (PEcAn.utils::status.check("MODEL") == 0) {
+  PEcAn.utils::status.start("MODEL")
+  stop_on_error <- as.logical(settings[[c("run", "stop_on_error")]])
+  if (length(stop_on_error) == 0) {
+    # If we're doing an ensemble run, don't stop. If only a single run, we
+    # should be stopping.
+    if (is.null(settings[["ensemble"]]) ||
+        as.numeric(settings[[c("ensemble", "size")]]) == 1) {
+      stop_on_error <- TRUE
+    } else {
+      stop_on_error <- FALSE
+    }
+  }
+  PEcAn.remote::runModule.start.model.runs(settings, stop.on.error = stop_on_error)
+  PEcAn.utils::status.end()
+}
+
+# Get results of model runs
+if (PEcAn.utils::status.check("OUTPUT") == 0) {
+  PEcAn.utils::status.start("OUTPUT")
+  runModule.get.results(settings)
+  PEcAn.utils::status.end()
+}
+
+# Run ensemble analysis on model output.
+if ("ensemble" %in% names(settings)
+    && PEcAn.utils::status.check("ENSEMBLE") == 0) {
+  PEcAn.utils::status.start("ENSEMBLE")
+  runModule.run.ensemble.analysis(settings, TRUE)
+  PEcAn.utils::status.end()
+}
+
+# Run state data assimilation
+if ("state.data.assimilation" %in% names(settings)) {
+  if (PEcAn.utils::status.check("SDA") == 0) {
+    PEcAn.utils::status.start("SDA")
+    settings <- sda.enfk(settings)
+    PEcAn.utils::status.end()
+  }
+}
+
+# Pecan workflow complete
+if (PEcAn.utils::status.check("FINISHED") == 0) {
+  PEcAn.utils::status.start("FINISHED")
+  PEcAn.remote::kill.tunnel(settings)
+  db.query(
+    paste(
+      "UPDATE workflows SET finished_at=NOW() WHERE id=",
+      settings$workflow$id,
+      "AND finished_at IS NULL"
+    ),
+    params = settings$database$bety
+  )
+  
+  # Send email if configured
+  if (!is.null(settings$email)
+      && !is.null(settings$email$to)
+      && (settings$email$to != "")) {
+    sendmail(
+      settings$email$from,
+      settings$email$to,
+      paste0("Workflow has finished executing at ", base::date()),
+      paste0("You can find the results on ", settings$email$url)
+    )
+  }
+  PEcAn.utils::status.end()
+}
+
+db.print.connections()
+print("---------- PEcAn Workflow Complete ----------")
+
+
+#EFI Output Configuration
```

Comment:
1. the goal of this remaining section of code will be unclear to most users
2. if the goal is to take output files that are in PEcAn standard and convert them to EFI's ensemble CSV standard, I'd make that into a function and put it into an appropriate package (utils?)
3. what's the difference between the PEcAn standard and the EFI (preferred) netcdf standard? in other words, is there even a need for this code? And if there is, why not use the EFI netcdf standard instead of the csv standard? Maybe there should be an issue (and discussion on the next PEcAn call) about reconciling the differences between the PEcAn and EFI standards so that we can avoid this step altogether.

Reply:
So the primary reason I put this at the end of the workflow is so that if someone is running the workflow and wants to use the results of the model runs for statistical analysis or graphs you end up with a usable .csv. And so they do not need another script to do that. This is convenient for the EFI forecasts of course because it will already be in the EFI format but I thought it would be helpful in a broader sense too?

Reply:
I'm not saying that it's not useful, I'm saying you could make it more useful by (A) making it a function (B) documenting what that function aims to do and (C) use the preferred EFI standard (netCDF) rather than the tier 2 csv. The last change can come in a future PR

Reply:
Got it